### ISSUE TITLE
nixvim standalone broken after staging-next 2023-10-26

### ISSUE THREAD

**Issue by andrevmatos** [nix-community/nixvim#707]
**Nixpkgs channel**: master

## Description
nixpkgs just got a staging-next merge to master (NixOS/nixpkgs#263535), which seems to have broken nixvim. It maybe due to NixOS/nixpkgs#237425 making `pkgs.neovim` not a `symlinkJoin` anymore, while nixvim standalone seems to rely on `paths` attribute here:
https://github.com/nix-community/nixvim/blob/a304b68a882357a55c4d76eb6db299e4d11780c5/wrappers/standalone.nix#L34-L35

```
error:
       … while calling the 'getAttr' builtin

         at /builtin/derivation.nix:19:19: (source not available)

       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'neovim-20231115_d92dd2a'
         whose name attribute is located at /nix/store/kjg0gm2y4f5zgw6cqidh4w6c5n3q6fww-source/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'paths' of derivation 'neovim-20231115_d92dd2a'

         at /nix/store/1ann4n8aba6qd5sr93gx1yjj7pinfk2d-source/wrappers/standalone.nix:34:5:

           33|   config.finalPackage.overrideAttrs (oa: {
           34|     paths =
             |     ^
           35|       oa.paths

       error: attribute 'paths' missing

       at /nix/store/1ann4n8aba6qd5sr93gx1yjj7pinfk2d-source/wrappers/standalone.nix:35:7:

           34|     paths =
           35|       oa.paths
             |       ^
           36|       ++ (pkgs.lib.optional config.enableMan self.packages.${pkgs.system}.man-docs);
```

cc @teto

**Comment by andrevmatos** [comment]
My workaround to make the `oa.paths` stop erroring, is to overlay `wrapNeovimUnstable`, used to create `finalPackage` before standalone:
```nix
final: prev: {
  wrapNeovimUnstable = pkg: args: (prev.wrapNeovimUnstable pkg args).overrideAttrs (_: {paths = [];});
}
```

**Comment by GaetanLepage** [comment]
Thank you for having reported this.
It is very annoying indeed. Do you think a similar solution would be made directly into the `nixvim` code ?
Do you have any suggestion ?

**Comment by teto** [comment]
I might be able to make more suggestions later, quite busy right now but the changes in nixpkgs should make the derivation more composable, as in you dont have to pass the definie correct configuration at the start but you can chain them. So instead of calling wrapNeovimUnstable, it's possible you can just do `neovim.overrideAttrs(oa: { ...})`

**Comment by traxys** [comment]
I have a fix:
```
diff --git a/wrappers/standalone.nix b/wrappers/standalone.nix
index 670921e5..6e62ecc9 100644
--- a/wrappers/standalone.nix
+++ b/wrappers/standalone.nix
@@ -31,7 +31,12 @@ default_pkgs: {
   config = handleAssertions eval.config;
 in
   config.finalPackage.overrideAttrs (oa: {
-    paths =
-      oa.paths
-      ++ (pkgs.lib.optional config.enableMan self.packages.${pkgs.system}.man-docs);
+    preInstall =
+      if config.enableMan
+      then ''
+        mkdir -p $out/share/man/man5
+        cp ${self.packages.${pkgs.system}.man-docs}/share/man/man5/nixvim.5 $out/share/man/man5
+      ''
+      else ''
+      '';
   })
```

But I won't be able to apply it right now, we first need to wait for the PR to [trickle down into nixos/nixpkgs-unstable](https://nixpk.gs/pr-tracker.html?pr=263535)

**Comment by GaetanLepage** [comment]
The update has landed. I guess that we should test and deploy this fix ASAP.

**Comment by traxys** [comment]
Should be fixed with #710

**Comment by andrevmatos** [comment]
Yes, it's fixed there.
It'd be great to move to `neovim.overrideAttrs` approach, as @teto suggested, but for now, it's working fine with `wrapNeovimUnstable`. Thanks
